larq_compute_engine/core/packbits_aarch64.h

```diff
@@ -10,62 +10,42 @@
 namespace compute_engine {
 namespace core {
 
-// This will bitpack exactly 64 floats.
-// It will be packed in a weird order which is described in the comments
+// This will bitpack the sign-bits from 64 floats, in a deterministic but
+// non-consecutive order.
 inline void packbits_aarch64_64(const float* input, std::uint64_t* output) {
   asm volatile(
-      "ld1    {v0.4s, v1.4s, v2.4s, v3.4s}, [%0], #64    \n"
-      "sri    v0.4s, v2.4s, #1    \n"
-      "sri    v1.4s, v3.4s, #1    \n"
-      "ld1    {v2.4s, v3.4s, v4.4s, v5.4s}, [%0], #64    \n"
-      "sri    v2.4s, v4.4s, #1    \n"
```

Comment:
I believe that this load + shift instruction pair isn't being pipelined because operands of the load are used in the shift. Likewise for all remaining load + shift pairs in the existing implementation.

Reply:
Nice find!